_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -57,6 +57,23 @@ default cmake.build_dir             {${workpath}/build}
 # minimal/initial value for the install rpath:
 default cmake.install_rpath         ${prefix}/lib
 
+# CMake provides several different generators corresponding to different utities
+# (and IDEs) used for building the sources. We support "Unix Makefiles" (the default)
+# and Ninja, a leaner-and-meaner alternative.
+#
+# In the Portfile, use
+#
+# cmake.generator Ninja
+# or
+# cmake.generator Unix Makefiles    # no quotes!
```

Comment:
All arguments to `option`s, such as `cmake.generator` are lists. This happens because an option is actually a `proc` and that `proc` accepts a variable number of arguments by naming its last parameter `args`. So when you write

```tcl
cmake.generator Unix Makefiles
```

what actually happens is that the `proc` that sets `cmake.generator` is called with a list with two elements, `Unix` and `Makefile`, or in Tcl formatting: `{Unix Makefiles}`.

If you make that `cmake.generator "Unix Makefiles"`, it will still be called with a list, but this time the list only has a single element: `{"Unix Makefiles"}`.

So, what I'd do in this case:
```tcl
proc cmake::handle_generator {option action args} {
  # ...
  if {[llength $args] > 1 || [llength $args] <= 0} {
    error "cmake.generator is a single-valued field and a value is required".
  }
  switch -nocase [lindex $args 0] {
    "Unix Makefiles" {
      # ...
    }
    "Ninja" {
      # ...
    }
    # ...
  }
  # ...
}
```